crates/script/src/broadcast.rs

```diff
@@ -160,14 +156,17 @@ pub struct BundledState {
 
 impl BundledState {
     pub async fn wait_for_pending(mut self) -> Result<Self> {
+        let progress = ScriptProgress::default();
+        let progress_ref = &progress;
```

Comment:
nit: I don't think this tmp var is needed?

Reply:
if using just `progress` in closure it tries to move the `progress` var itself into the closure and fails to compile

```
error[E0507]: cannot move out of `progress`, a captured variable in an `FnMut` closure
   --> crates/script/src/broadcast.rs:165:45
    |
159 |           let progress = ScriptProgress::default();
    |               -------- captured outer variable
...
165 |               .map(|(sequence_idx, sequence)| async move {
    |  __________________--------------------------_^
    | |                  |
    | |                  captured by this `FnMut` closure
166 | |                 let rpc_url = sequence.rpc_url();
167 | |                 let provider = Arc::new(get_http_provider(rpc_url));
168 | |                 (&progress).wait_for_pending(sequence_idx, sequence, &provider).await
    | |                   --------
    | |                   |
    | |                   variable moved due to use in coroutine
    | |                   move occurs because `progress` has type `ScriptProgress`, which does not implement the `Copy` trait
169 | |             })
    | |_____________^ `progress` is moved here
```